app/config/services.php

```diff
@@ -18,6 +23,24 @@
  */
 
 $container = new ContainerBuilder();
```

Comment:
Ja, DI (Dependency Injection) werkt zoals je het beschrijft. Service zijn afhankelijk van ander Service. Als je een Service nodig heb, dan Inject je deze.

Je voorbeeld is niet een goed voorbeeld. Want ik zie de Mail_Transport. Die wil ik bijvoorbeeld ook in mijn Service hebben. Dat kan nu niet. Inject de Transport via de Constructor.

Je heb tevens SOLID. Een klasse maakt eigenlijk geen 'new instances' van een klasse meer aan. Je Inject ze.